src/rest/resource.cpp

```diff
@@ -489,6 +499,75 @@ void Resource::Rloc(const Request &aRequest, Response &aResponse) const
     }
 }
 
+void Resource::GetActiveDatasetTlvs(Response &aResponse) const
+{
+    otOperationalDatasetTlvs datasetTlvs;
+    otError                  error        = OT_ERROR_NONE;
+    std::string    body;
+    std::string    errorCode;
+
+    SuccessOrExit(otDatasetGetActiveTlvs(mInstance, &datasetTlvs));
+
+    body     = Json::Bytes2HexJsonString(datasetTlvs.mTlvs, datasetTlvs.mLength);
+
+    aResponse.SetBody(body);
+    errorCode = GetHttpStatus(HttpStatusCode::kStatusOk);
+    aResponse.SetResponsCode(errorCode);
+
+exit:
+    if (error != OT_ERROR_NONE)
+    {
+        otbrLogWarning("Failed to get active dataset: %s", otThreadErrorToString(error));
+        ErrorHandler(aResponse, HttpStatusCode::kStatusInternalServerError);
+    }
+}
+
+void Resource::SetActiveDatasetTlvs(const Request &aRequest, Response &aResponse) const
+{
+    int ret;
+    otOperationalDatasetTlvs datasetTlvs;
+    otError                  error = OT_ERROR_NONE;
+    std::string    errorCode;
+
+    ret = Json::Hex2BytesJsonString(aRequest.GetBody(), datasetTlvs.mTlvs, OT_OPERATIONAL_DATASET_MAX_LENGTH);
+    if (ret < 0)
+    {
+        errorCode = GetHttpStatus(HttpStatusCode::kStatusBadRequest);
+        aResponse.SetResponsCode(errorCode);
+        return;
```

Comment:
```suggestion
        ExitNow();
```

Reply:
This would overwrite the HTTP return code...

Reply:
The entire `exit:` block is conditional on `error != OT_ERROR_NONE`. What am I missing?

Reply:
Right!